fleet-be/src/main/java/com/everis/training/fleet/business/fleet/control/VehicleController.java

```diff
@@ -22,27 +23,45 @@
         return repo.getAllFreeVehicles();
     }
 
-    public void add(Vehicle vehicle) {
-        if (vehicle.getVin().length()==17){
-            repo.saveVehicle(vehicle);
+    public void delete(String vin) throws Exception{
+        try {
+            vinCheck(vin);
+            repo.deleteVehicle(vin);
+        } catch (Exception e) {
+            e.printStackTrace();
         }
     }
 
-    public void delete(String vin) {
-        if (vin.length() == 17){
-            repo.deleteVehicle(vin);
+    public void reserveCar(Integer id, String vin) throws Exception{
+        try {
+            vinCheck(vin);
+            repo.reserveCar(id, vin);
+        } catch (Exception e) {
+            e.printStackTrace();
         }
-
     }
 
-    public void update(Vehicle vehicle){
-        if (vehicle.getVin().length()==17){
-            repo.updateVehicle(vehicle);
+    public void add(Vehicle vehicle) throws Exception {
+        try {
+            vinCheck(vehicle.getVin());
+            repo.addVehicle(vehicle);
+        } catch (Exception e) {
+            throw e;
         }
     }
 
     public Vehicle findVehicle(String vin){
         return repo.findVehicle(vin);
     }
 
+    public void vinCheck (String vin) throws Exception{
+        int vinLength = 17;
+        if (vin==null){
+            throw new NullPointerException("The vin can not be null");
```

Comment:
All developers hate NPE's ;-). Please, use another exception more accurate like IllegalArgumentException or even refactor your VinLengthException as InvalidVinException and use the proper message when the vin is not informed.